components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -228,12 +232,13 @@ export const ChannelModerationTableRow: React.FunctionComponent<RowProps> = (pro
 export default class ChannelModeration extends React.Component<Props> {
     private errorMessagesToDisplay = (entry: ChannelPermissions): Array<any> => {
         const errorMessages: Array<any> = [];
-        const isGuestsDisabled = !isNil(entry.roles.guests?.['enabled']) && !entry.roles.guests?.['enabled'];
+        const isGuestsDisabled = !isNil(entry.roles.guests?.['enabled']) && !entry.roles.guests?.['enabled'] && this.props.guestAccountsEnabled;
         const isMembersDisabled = !entry.roles.members.enabled;
         let createPostsKey = '';
         if (entry.name === Permissions.CHANNEL_MODERATED_PERMISSIONS.USE_CHANNEL_MENTIONS) {
             const createPostsObject = this.props.channelPermissions && this.props.channelPermissions!.find((permission) => permission.name === Permissions.CHANNEL_MODERATED_PERMISSIONS.CREATE_POST);
-            if (!createPostsObject!.roles.guests!.value && !createPostsObject!.roles.members!.value) {
+            const createPostsGuestsUnchecked = !createPostsObject!.roles.guests!.value && this.props.guestAccountsEnabled;
+            if (createPostsGuestsUnchecked && !createPostsObject!.roles.members!.value) {
```

Comment:
Is there a reason you split this check into two separate lines? ie. Checking the member and guest roles on separate lines.

Reply:
@sbishel Nope! I just thought it look cleaner but I can revert it. I have no preference.